### PR TITLE
Eating animation

### LINKED ISSUE #45: Implement UC_09 eat
As soon as the user buys a food from the shop, it is given to the pet. Then happiness is updated (in future implementations).
Implement also the animations.

### PR DESCRIPTION
**Pull request by YuanjenHung**

### CODE REVIEW

**Comment by edogab33** [comment]
Is the animation triggered when the user buys an item or is that still to be implemented?

**Comment by YuanjenHung** [comment]
No, I don't think so, 

because the eating animation is not merged into main yet, and at the same moment Simone was developing the logic behind food button, so the animation trigger part is not yet implement. We are waiting you to merge everything, so we can do some integration test.

**Comment by edogab33** [comment]
> No, I don't think so,
> 
> because the eating animation is not merged into main yet, and at the same moment Simone was developing the logic behind food button, so the animation trigger part is not yet implement. We are waiting you to merge everything, so we can do some integration test.

The trigger of the animation should be done in this pull request, since you have all the code already available to do it. After that I will test on my machine everything, make the code review and hopefully merge the request if everything is working fine

**Comment by edogab33** [comment]
> No, I don't think so,
> 
> because the eating animation is not merged into main yet, and at the same moment Simone was developing the logic behind food button, so the animation trigger part is not yet implement. We are waiting you to merge everything, so we can do some integration test.

Alright, I saw that you have two different merge request for the same UC.

It was better to do it in one shot so you could have worked in parallel. Tomorrow I will test this implementation and hopefully merge it.

File: app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java [hunk 1]
```diff
@@ -31,17 +34,25 @@
...
+import java.util.ArrayList;
 
 public class HomeFragment extends Fragment {
 
     private FragmentHomeBinding binding;
     private RelativeLayout chic_walking_area;
-    private GifImageView chic;
-    private int x_chic, y_chic;
-    private int next_x_direction, next_y_direction;
+    private GifImageView chic, heart;
```
Comment (edogab33) [inline app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java:43]: In my environment GifImageView is not found. Are you sure that all the imports are correct? Does it work in yours?

File: app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java [hunk 2]
```diff
@@ -31,17 +34,25 @@
...
+    private LinearLayout food_linearLayout;
+    private ArrayList<ImageView> foods;
+    private RelativeLayout.LayoutParams lp_chic, lp_heart;
+    private int x_chic, y_chic, next_x_direction, next_y_direction;
     private int animation_frame, animation_frameSet_size;
+    private String chic_state, food_selected;
+
```
Comment (edogab33) [inline app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java:50]: Avoid useless spaces

File: app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java [hunk 3]
```diff
@@ -96,37 +123,144 @@ public void startAnimation(RelativeLayout area, GifImageView chic) {
...
-                            next_y_direction = new Random().nextInt(3) + -1;
-                            // constrain chic walking area, if chic is out of range, then change direction
-                            if (x_chic > 190) next_x_direction = -1;
-                            if (x_chic < 50) next_x_direction = 1;
-                            if (y_chic > 110) next_y_direction = -1;
-                            if (y_chic < 50) next_y_direction = 1;
-                            // horizontally flip the img based on direction
-                            if (next_x_direction != 0) chic.setScaleX(-next_x_direction);
-                        }
+                        if (chic_state == "idle") {
```
Comment (edogab33) [inline app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java:126]: You will need to comment and document better this part of code.

e.g. Describe why you need to check the chick state.

However it is enough to do it after the 20th and before the 23rd.

**Comment by edogab33** [comment]
Well documented code overall, although I couldn't test the eating animation and there is that problem with the GifImageView object

**Pull request merged**